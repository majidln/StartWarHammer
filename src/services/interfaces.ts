export interface Ingredient {
  carbs?: Number;
  fat?: Number;
  protein?: Number;
  fiber?: Number;
  calories?: Number;
}

export interface Nutrition {
  values: Ingredient;
  percentages: Ingredient;
}

export interface Tag {
  id: String;
  type: String;
  title: String;
}

export interface Instruction {
  title: String;
  steps: Array<String>;
  header?: Object;
  footer?: Object;
}

export interface Images {
  hz: String;
  vt: String;
  brightness: String;
}

export interface Recipe {
  id: string;
  slug: String;
  title: string;
  description: string;
  nutrition: Nutrition;
  time: Object;
  difficulty: Object;
  images: Images;
  tags: Array<Tag>;
  instructionSections: Array<Instruction>;
  rating: Number;
  isMembersOnly: Boolean;
  modifiedAt: String;
}

export interface RecipePage {
  listRecipes: Array<Recipe>;
  totalSize: Number;
  nextPage: Number;
}

export interface QueryResponse {
  data: RecipePage;
  loading: Boolean;
  fetchMore: Function;
}
