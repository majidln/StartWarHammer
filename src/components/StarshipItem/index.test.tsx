import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import {render, waitFor} from '@testing-library/react-native';
import {StarshipItem} from './index';

import {Starship} from '@services/interfaces';

const starship: Starship = {
  name: 'A-wing',
  id: 'cj0nwtqudq53b0114s5id3anc',
  crew: 1,
  length: 9.6,
  films: [
    {
      id: 'cj0nxmy5ega600114uiw05wle',
      title: 'Return of the Jedi',
    },
  ],
  __typename: 'Starship',
};

it('Starship Item Snapshot renders correctly', () => {
  const tree = renderer.create(<StarshipItem starship={starship} />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('Starship Item render in list correctly', async () => {
  const {findByTestId} = render(<StarshipItem starship={starship} />);

  await waitFor(() => findByTestId('starshipId'));
});
