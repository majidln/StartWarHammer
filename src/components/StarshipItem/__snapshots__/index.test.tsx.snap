// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Starship Item Snapshot renders correctly 1`] = `
<View
  testID="starshipId"
>
  <View
    style={
      Object {
        "flexDirection": "row",
        "flexWrap": "wrap",
        "justifyContent": "center",
        "marginVertical": 4,
        "width": "100%",
      }
    }
    testID="keyValueId"
  >
    <Text
      style={
        Object {
          "color": "#2c3e50",
          "fontSize": 18,
          "paddingRight": 4,
        }
      }
    >
      Crew:
    </Text>
    <Text
      style={
        Object {
          "color": "#000",
          "fontSize": 18,
          "fontWeight": "bold",
        }
      }
    >
      1
    </Text>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
        "flexWrap": "wrap",
        "justifyContent": "center",
        "marginVertical": 4,
        "width": "100%",
      }
    }
    testID="keyValueId"
  >
    <Text
      style={
        Object {
          "color": "#2c3e50",
          "fontSize": 18,
          "paddingRight": 4,
        }
      }
    >
      Length:
    </Text>
    <Text
      style={
        Object {
          "color": "#000",
          "fontSize": 18,
          "fontWeight": "bold",
        }
      }
    >
      9.6
    </Text>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
        "flexWrap": "wrap",
        "justifyContent": "center",
        "marginVertical": 4,
        "width": "100%",
      }
    }
    testID="keyValueId"
  >
    <Text
      style={
        Object {
          "color": "#2c3e50",
          "fontSize": 18,
          "paddingRight": 4,
        }
      }
    >
      Exist in:
    </Text>
    <Text
      style={
        Object {
          "color": "#000",
          "fontSize": 18,
          "fontWeight": "bold",
        }
      }
    >
      Return of the Jedi
    </Text>
  </View>
</View>
`;
